app/controllers/application_controller.rb

```diff
@@ -20,4 +20,13 @@ def current_user
   def redirect_to_home
     redirect_to root_path
   end
+
+  def require_redirection?
+    case action_name
+    when 'destroy'
+      return true unless logged_in?
+    else
+      return true if logged_in?
```

Comment:
How about

```
def xx?
  return true if logged_in? && action_name != 'destroy'

  false
end
```

Reply:
Fixed in b0f5d7a98e9434e3a6ae272b9d2de662e8bbc2bd